lib/portage/eapi.py

```diff
@@ -2,33 +2,54 @@
 # Distributed under the terms of the GNU General Public License v2
 
 import collections
-import operator
-import types
-
-from portage import eapi_is_supported
+from functools import lru_cache
 
 
+@lru_cache(None)
 def eapi_has_iuse_defaults(eapi):
+    if eapi is None:
+        return True
+
     return eapi != "0"
```

Comment:
To be honest, I don't like the added verbosity. How about covering the `None` in the actual  check? e.g.:
```suggestion
    return eapi not in ("0",)
```

Reply:
My only worry is that readers will be confused by the presence of `None` in some of these functions but not others. The fact that `eapi` can be `None` is strange to begin with.

Reply:
This has been reworked, @mgorny @mattst88 please take a second look and let me know your thoughts.